### PR TITLE
Add lodash cloneDeep to support arrays in omitPaths

### PR DESCRIPTION
**Pull request by MiguelNazMor**
… and unit test with the case.

(Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.)

What does this implement/fix? Explain your changes.
----------------------------------------------------

While working with the middleware I discovered that, when adding a path to omitPaths which is apart of an array ("response.users[0].email"), the field was deleted form the log but also from the response.

Using cloneDeep fixes this behaviour (unit test added).

Does this close any currently open issues?
------------------------------------------
…

Any relevant logs, error output, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

Any other comments?
-------------------
…

Where has this been tested?
---------------------------
**Node.js Versions:** …
--12.4--
**Middy Versions:** …

**AWS SDK Versions:** …

Todo list
---------

[ X] Feature/Fix fully implemented
[ X] Added tests
[ NA] Updated relevant documentation
[ NA] Updated relevant examples

### CODE REVIEW

File: packages/input-output-logger/index.js [hunk 1]
```diff
@@ -9,7 +10,8 @@ module.exports = (opts) => {
   const { logger, omitPaths } = Object.assign({}, defaults, opts)
 
   const omitAndLog = message => {
-    const redactedMessage = omit(message, omitPaths)
+    const messageClone = cloneDeep(message)
```
Comment (willfarrell) [inline packages/input-output-logger/index.js:13]: Can you test using `JSON.parse(JSON.stringify(o))` instead. `_.cloneDeep` is notoriously slow in comparison, unless you're using the additional edge cases.

https://github.com/lodash/lodash/issues/1984

Comment (MiguelNazMor) [reply packages/input-output-logger/index.js]: Updated and tested, thanks for your comment!
